test/unit/lib/github_basic_detective_test.rb

```diff
@@ -29,7 +31,34 @@ class GithubBasicDetectiveTest < ActiveSupport::TestCase
 
       assert results.key?(:license)
       assert results[:license].key?(:value)
-      assert results[:license][:value] == 'MIT'
+      assert_equal 'MIT', results[:license][:value]
+
+      # Do several unit tests of language_cleanup, it's more complex.
+      assert_equal '', detective.language_cleanup({})
+      assert_equal 'C', detective.language_cleanup(C: 1000) # singleton
+      # We need to use the older hash syntax to spell some names
+      # rubocop:disable Style/HashSyntax
+      assert_equal 'C++, C', detective.language_cleanup( # actually sorts
+        C: 1000, :"C++" => 5000
+      )
+      assert_equal 'C#, JavaScript', detective.language_cleanup(
+        HTML: 1000, :"C#" => 5000, JavaScript: 800
+      )
+      # This is the list from https://api.github.com/repos/
+      # assimilation/assimilation-official/languages
+      assert_equal 'Python, C, Shell, C++, CMake, C#, Ruby',
+                   detective.language_cleanup(
+                     Python: 1_151_127,
+                     C: 1_059_779,
+                     Shell: 285_358,
+                     :"C++" => 44_086,
+                     CMake: 33_855,
+                     :"C#" => 29_834,
+                     HTML: 2_290,
+                     Ruby: 1_359
+                   )
```

Comment:
This seems like a bad thing to build into a test, since the values will change the next time the cassette is re-recorded, right?

Could you instead run on a known value that won't change, like the values of a specific release: https://github.com/kubernetes/kubernetes/tree/release-1.0